ansible_collections/arista/avd/tests/unit/test/test_defined_plugin.py

```diff
@@ -0,0 +1,117 @@
+from ansible_collections.arista.avd.plugins.test.defined import defined, TestModule
+from jinja2.runtime import Undefined
+from ansible.errors import AnsibleError
+import pytest
+
+VALUE_LIST = ['ab', None, 1, True]
+TEST_VALUE_LIST = [None, 'ab', True, 1, True]
+FAIL_ACTION_LIST = ['warning', 'error']
+VAR_NAME_LIST = ['ab', None, 1]
+VAR_TYPE_LIST = ['int', 'str', 'integer', 'aaa', None]
+INVALID_FAIL_ACTION_LIST = [None, 'aaaa']
+
+f = TestModule()
+
+
+class TestDefinedPlugin():
+
+    def defined_function(self, value, test_value=None, var_type=None, fail_action=None, var_name=None, err_msg=None, warn_msg=None):
+        if str(fail_action).lower() == 'warning':
+            resp, warning = defined(value, test_value=test_value, var_type=var_type, fail_action=fail_action,
+                                    var_name=var_name, run_tests=True)
+            if warn_msg:
+                assert warning is not None
+                warn = str(list(warning.keys())[0]).replace(
+                    '[WARNING]: ', '').strip().replace('\n', ' ')
+                assert warn == warn_msg
+                assert resp is False
+        elif str(fail_action).lower() == 'error':
+            with pytest.raises(AnsibleError) as e:
+                resp, warning = defined(value, test_value=test_value, var_type=var_type, fail_action=fail_action,
+                                        var_name=var_name, run_tests=True)
+            assert str(e.value) == err_msg
+
+    @pytest.mark.parametrize("VALUE", VALUE_LIST)
+    @pytest.mark.parametrize("FAIL_ACTION", FAIL_ACTION_LIST)
+    @pytest.mark.parametrize("VAR_NAME", VAR_NAME_LIST)
+    def test_defined_plugin_value_undefined_or_none(self, VALUE, FAIL_ACTION, VAR_NAME):
+        if isinstance(VALUE, Undefined) or VALUE is None:
+            if str(FAIL_ACTION).lower() == 'warning':
+                if VAR_NAME is not None:
+                    warn_msg = f"{VAR_NAME} was expected but not set. Output may be incorrect or incomplete!"
+                else:
+                    warn_msg = "A variable was expected but not set. Output may be incorrect or incomplete!"
+
+                self.defined_function(
+                    VALUE, fail_action=FAIL_ACTION, var_name=VAR_NAME, warn_msg=warn_msg)
+            elif str(FAIL_ACTION).lower() == 'error':
+                if VAR_NAME is not None:
+                    err_msg = f"{VAR_NAME} was expected but not set!"
+                else:
+                    err_msg = "A variable was expected but not set!"
+
+                self.defined_function(
+                    VALUE, fail_action=FAIL_ACTION, var_name=VAR_NAME, err_msg=err_msg)
+
+    @pytest.mark.parametrize("VALUE", VALUE_LIST)
+    def test_defined_plugin_value_not_none_and_define(self, VALUE):
```

Comment:
I think this test is covered in the last scenario ```test_defined_plugin_fail_action_None```, for TEST_VALUE = None and INVALID_FAIL_ACTION = None. But it's ok.